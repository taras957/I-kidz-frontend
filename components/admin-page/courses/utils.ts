import { translationsType } from 'api/course/interfaces/course';
import * as yup from 'yup';

export const getSchema = (lang: translationsType) =>
  yup.object().shape({
    translations: yup.object().shape({
      [lang]: yup.object().shape({
        title: yup
          .string()
          .required(`це поле є обов'язковим`)
          .min(5, 'мінімум 5 символів')
          .max(40, 'максимум 40 символів'),
        subtitle: yup
          .string()
          .min(5, 'мінімум 5 символів')
          .required("це поле є обов'язковим"),
        price: yup
          .string()
          .min(3, 'мінімум 3 символи')
          .required("це поле є обов'язковим"),
        duration: yup
          .string()
          .min(5, 'мінімум 5 символів')
          .required("це поле є обов'язковим"),
        description: yup
          .string()
          .min(5, 'мінімум 5 символів')
          .required("це поле є обов'язковим"),
      }),
    }),
    category: yup.string().required("це поле є обов'язковим"),
    picture: yup
      .mixed()
      .required('You need to provide a file')
      .test('fileSize', 'File Size is too large', (value) => {
        if (value) {
          return value.size <= 5242880;
        }
        return false;
      }),
  });

export const getDefaultValues = (lang: translationsType) => {
  return {
    translations: {
      [lang]: {
        title: '',
        subtitle: '',
        price: '',
        duration: '',
        description: '',
      },
    },
    age: '',
    category: '',
    picture: null,
    is_active: true,
  };
};
