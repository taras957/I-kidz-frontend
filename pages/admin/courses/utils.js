const getUnique = (list) => {
  const uniqueCategories = list.reduce((acc, node) => {
    acc[node.category] = node.age;
    return acc;
  }, {});
  return uniqueCategories;
};
export const getSorted = (list) => {
  const unique = getUnique(list);
  const copy = Object.entries(unique);
  copy.sort((a, b) => {
    const matchesA = a[1].match(/(\d+)/);
    const matchesB = b[1].match(/(\d+)/);
    return matchesA[0] - matchesB[0];
  });
  return copy;
};