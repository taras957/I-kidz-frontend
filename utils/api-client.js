const apiURL = process.env.NEXT_PUBLIC_API;
const apiPrefix = process.env.NEXT_PUBLIC_API_PREFIX;
import axios from "axios";
axios.defaults.withCredentials = true

async function client(
  endpoint,
  { data, token, headers: customHeaders, isBlob = false, ...customConfig } = {}
) {
  const config = {
    method: data ? "POST" : "GET",
    data: data ? (isBlob ? data : JSON.stringify(data)) : undefined,
    url: `${apiURL}/${apiPrefix}/${endpoint}`,

    headers: {
      Authorization: token ? `Bearer ${token}` : undefined,
      "Content-Type": data ? "application/json" : undefined,
      ...customHeaders,
    },
    ...customConfig,
  };

  return await axios(config)
    .then((response) => {
      // if (response.status === 401) {
      //   queryCache.clear()
      //   await auth.logout()
      //   // refresh the page for them
      //   window.location.assign(window.location)
      //   return Promise.reject({message: 'Please re-authenticate.'})
      // }
      return response;
    })
    .catch(function (error) {
      // throw new Error(error)
      return "error";
    });
}
export { client };
