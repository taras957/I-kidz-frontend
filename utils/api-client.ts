import axios, { Method } from 'axios';
const apiURL = process.env.NEXT_PUBLIC_API;
const apiPrefix = process.env.NEXT_PUBLIC_API_PREFIX;

interface IAxiosConfig {
  method: Method;
  data: unknown;
  url: string;
  headers: unknown;
}

interface IClientOptions {
  data?: unknown;
  token?: string;
  headers?: Object;
  isBlob?: boolean;
}

async function client<T>(
  endpoint: string,
  {
    data,
    token,
    headers: customHeaders,
    isBlob = false,
    ...customConfig
  }: IClientOptions = {}
) {
  const config: IAxiosConfig = {
    method: data ? 'POST' : 'GET',
    data: data ? (isBlob ? data : JSON.stringify(data)) : undefined,
    url: `${apiURL}/${apiPrefix}/${endpoint}`,

    headers: {
      'Content-Type': data ? 'application/json' : undefined,
      ...customHeaders,
    },
    ...customConfig,
  };

  if (token) {
    Object.assign(config.headers, { Authorization: `Bearer ${token}` });
  }
  return axios
    .request<T>(config)
    .then((response) => {
      // if (response.status === 401) {
      //   queryCache.clear()
      //   await auth.logout()
      //   // refresh the page for them
      //   window.location.assign(window.location)
      //   return Promise.reject({message: 'Please re-authenticate.'})
      // }
      return response;
    })
    .catch((e: unknown) => {
      throw new Error(e.message);
      // if (typeof e === 'string') {
      //   e.toUpperCase();
      // } else if (e instanceof Error) {
      //   e.message;
      // }
    });
}
export { client };
