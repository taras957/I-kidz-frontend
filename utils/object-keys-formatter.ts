export class ObjectKeysTransformation {
  // !TODO implement formatter from snail to Camel case
  toCamelCase() {
    console.log('todo');
  }

  toSnailCase<T>(values: T) {
    for (const key in values) {
      let shouldDelete = false;
      const snakeCaseKey = key.replace(/[A-Z]/g, (letter, index) => {
        shouldDelete = true;
        return index == 0 ? letter.toLowerCase() : '_' + letter.toLowerCase();
      });
      Object.assign(values, { [snakeCaseKey]: values[key] });
      if (shouldDelete) {
        delete values[key];
        shouldDelete = false;
      }
    }
    return values;
  }
}
