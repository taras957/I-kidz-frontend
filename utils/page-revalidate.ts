import axios from 'axios';
const revalidateToken = process.env.NEXT_PUBLIC_MY_REVALIDATE_TOKEN;
// const
export const revalidate = (path = '/') => {
  return axios(`/api/revalidate?secret=${revalidateToken}&path=${path}`);
};
