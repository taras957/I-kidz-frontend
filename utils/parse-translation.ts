type languages = 'ua' | 'rus' | 'eng';
interface ITranslationParse<T> {
  translations: {
    [key in languages]: T;
  };
}
export function parseTranslation<T extends ITranslationParse<K>, K>(
  list: T[],
  language: languages
) {
  return list.map(({ translations, ...course }) => ({
    ...course,
    ...translations[language],
  }));
}
