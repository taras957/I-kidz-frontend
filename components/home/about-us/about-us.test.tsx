import { screen } from '@testing-library/react';
import AboutUs from '@/components/home/about-us';
import { render } from '__tests__/utils/index';

describe('testing about us component', () => {
  test(' to be in document', async () => {
    render(<AboutUs />);

    const section = screen.queryByTestId('about-us');
    expect(section).toBeInTheDocument();
    expect(section).toMatchSnapshot();
  });
});
