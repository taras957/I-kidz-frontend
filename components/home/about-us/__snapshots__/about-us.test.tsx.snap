// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing about us component  to be in document 1`] = `
<section
  class="root"
  data-testid="about-us"
>
  <div
    class="side-pic"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTEzIiBoZWlnaHQ9IjQ4NCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="girl with laptop"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
  <div
    class="side-content"
  >
    <h2
      class="title"
    >
      Про Нас
    </h2>
    <p
      class="subtitle"
    >
      iKIDZ – це цифровий простір для розвитку технологічних навичок у дітей. В iKIDZ діти вивчають сучасні мови програмування, комп’ютерні технології, математику та англійську мову. Наша унікальна методика:
    </p>
    <ul
      class="pros-list"
    >
      <li
        class="pros-item"
      >
        Навчання через практику: діти навчаються цифровій грамотності, ІТ-розробці, кодингу, дизайну у команди професійних ІТ спеціалістів під патронажем ІТ компаній.
      </li>
      <li
        class="pros-item"
      >
        Сучасні мови програмування: ми вчимо мови програмування та розвиваємо логіку та алгоритміку, що використовуються у 99% компаній у світі.
      </li>
      <li
        class="pros-item"
      >
        Розвиток логіки та креативу: діти навчаються думати та створювати, а не завчати напам’ять мови програмування.
      </li>
      <li
        class="pros-item"
      >
        Розвиток стартап мислення: діти створюють власні проекти та вчаться їх захищати. Беруть участь у різноманітних конкурсах, ІТ-хакатонах, олімпіадах та мітапах.
      </li>
      <li
        class="pros-item"
      >
        Заходи для батьків: цікаві зустрічі, сучасні івенти, подорожі та постійний зв’язок.
      </li>
      <li
        class="pros-item"
      >
        Ми разом йдемо до спільної мети і допомогаємо дитині реалізувати себе і стати успішною.
      </li>
    </ul>
    <p
      class="motto"
    >
      Наші діти не грають в комп’ютерні ігри, а створюють їх самі!
    </p>
  </div>
</section>
`;
